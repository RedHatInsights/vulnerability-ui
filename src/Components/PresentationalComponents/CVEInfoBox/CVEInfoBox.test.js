import CveInfoBox from './CVEInfoBox';
import toJson from 'enzyme-to-json';

const testTitle = 'testTitle';
const testDescription = 'testDescription';

describe('CveInfoBox', () => {
    it('Should render without params', () => {
        const component = mount(<CveInfoBox/>);
        expect(toJson(component)).toMatchSnapshot();
    });
    it('Should render with title only', () => {
        const component = mount(<CveInfoBox titleContent={testTitle}/>);
        expect(toJson(component)).toMatchSnapshot();
    });
    it('Should render with description only', () => {
        const component = mount(<CveInfoBox descriptionContent={testDescription}/>);
        expect(toJson(component)).toMatchSnapshot();
    });
    it('Should redner with both title and description', () => {
        const component = mount(<CveInfoBox titleContent={testTitle} descriptionContent={testDescription}/>);
        expect(toJson(component)).toMatchSnapshot();
    })
});