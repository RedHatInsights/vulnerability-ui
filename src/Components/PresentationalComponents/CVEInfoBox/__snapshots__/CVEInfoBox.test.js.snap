// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CveInfoBox Should redner with both title and description 1`] = `
<CVEInfoBox
  descriptionContent="testDescription"
  titleContent="testTitle"
>
  <Component
    className="infobox-square"
    hasGutter={true}
  >
    <div
      className="pf-l-split pf-m-gutter infobox-square"
    >
      <WithLoader
        loading={true}
      >
        <Skeleton
          isDark={false}
          size="lg"
        >
          <div
            className="ins-c-skeleton ins-c-skeleton__lg"
          >
             
          </div>
        </Skeleton>
      </WithLoader>
    </div>
  </Component>
</CVEInfoBox>
`;

exports[`CveInfoBox Should render with description only 1`] = `
<CVEInfoBox
  descriptionContent="testDescription"
>
  <Component
    className="infobox-square"
    hasGutter={true}
  >
    <div
      className="pf-l-split pf-m-gutter infobox-square"
    >
      <WithLoader
        loading={true}
      >
        <Skeleton
          isDark={false}
          size="lg"
        >
          <div
            className="ins-c-skeleton ins-c-skeleton__lg"
          >
             
          </div>
        </Skeleton>
      </WithLoader>
    </div>
  </Component>
</CVEInfoBox>
`;

exports[`CveInfoBox Should render with title only 1`] = `
<CVEInfoBox
  titleContent="testTitle"
>
  <Component
    className="infobox-square"
    hasGutter={true}
  >
    <div
      className="pf-l-split pf-m-gutter infobox-square"
    >
      <WithLoader
        loading={true}
      >
        <Skeleton
          isDark={false}
          size="lg"
        >
          <div
            className="ins-c-skeleton ins-c-skeleton__lg"
          >
             
          </div>
        </Skeleton>
      </WithLoader>
    </div>
  </Component>
</CVEInfoBox>
`;

exports[`CveInfoBox Should render without params 1`] = `
<CVEInfoBox>
  <Component
    className="infobox-square"
    hasGutter={true}
  >
    <div
      className="pf-l-split pf-m-gutter infobox-square"
    >
      <WithLoader
        loading={true}
      >
        <Skeleton
          isDark={false}
          size="lg"
        >
          <div
            className="ins-c-skeleton ins-c-skeleton__lg"
          >
             
          </div>
        </Skeleton>
      </WithLoader>
    </div>
  </Component>
</CVEInfoBox>
`;
